### PR TITLE
Reduce allocations to improve performance.

### PR DESCRIPTION
**Pull request by BumblingCoder**
This commit partially addresses the poor performance while playing the Skiessi-RandomProjectNumber demo song. This is basically only an issue because of issue  #3865, but decreasing the number of allocations is probably good regardless of allocator performance.

### CODE REVIEW

**Comment by BumblingCoder** [comment]
The Organic instrument has the same issue with excessive allocations as the TripleOsc instrument did. I'm adding it to this pull request with the same changes.

File: plugins/triple_oscillator/TripleOscillator.cpp [hunk 1]
```diff
@@ -326,33 +327,32 @@ void TripleOscillator::playNote( NotePlayHandle * _n,
...
-						&m_osc[i]->m_waveShapeModel,
-						&m_osc[i]->m_modulationAlgoModel,
-						_n->frequency(),
-						m_osc[i]->m_detuningRight,
-						m_osc[i]->m_phaseOffsetRight,
-						m_osc[i]->m_volumeRight,
-						oscs_r[i + 1] );
+				new (&oscs_l[i]) Oscillator(
+							&m_osc[i]->m_waveShapeModel,
+							&m_osc[i]->m_modulationAlgoModel,
```
Comment (Umcaruje) [inline plugins/triple_oscillator/TripleOscillator.cpp]: You have unnecessary indentation here

**Comment by tresf** [comment]
@softrabbit or @gi0e5b06 (or someone else) can you please review this?

File: include/Oscillator.h [hunk 2]
```diff
@@ -42,6 +42,10 @@ class IntModel;
 class EXPORT Oscillator
 {
 	MM_OPERATORS
+	void* operator new( std::size_t, void* p )
```
Comment (lukas-w) [inline include/Oscillator.h]: Why is this override needed? Isn't this the default implementation?

File: plugins/organic/organic.cpp [hunk 3]
```diff
@@ -233,8 +233,9 @@ void organicInstrument::playNote( NotePlayHandle * _n,
 	
 	if( _n->totalFramesPlayed() == 0 || _n->m_pluginData == NULL )
 	{
-		Oscillator * oscs_l[m_numOscillators];
-		Oscillator * oscs_r[m_numOscillators];
+		Oscillator * oscs_all = MM_ALLOC( Oscillator, m_numOscillators*2 );
```
Comment (lukas-w) [inline plugins/organic/organic.cpp]: Seeing that `Oscillator` has overloaded `new` operators through the `MM_OPERATORS` macro, we should be able to replace this with

```cpp
Oscillator* oscs_all = new Oscillator[m_numOscillators * 2]
```

File: plugins/organic/organic.cpp [hunk 4]
```diff
@@ -233,8 +233,9 @@ void organicInstrument::playNote( NotePlayHandle * _n,
...
+		Oscillator * oscs_l = oscs_all;
+		Oscillator * oscs_r = oscs_all + m_numOscillators;
```
Comment (lukas-w) [inline plugins/organic/organic.cpp]: Maybe this alternative is clearer to readers, but I'll leave it up to you
```cpp
Oscillator * oscs_l = & oscs_all[0];
Oscillator * oscs_r = & oscs_all[m_numOscillators];
```

File: plugins/organic/organic.cpp [hunk 5]
```diff
@@ -269,30 +270,30 @@ void organicInstrument::playNote( NotePlayHandle * _n,
...
+						&oscs_r[i + 1] );
 			}
 			
 				
 		}
 
-		static_cast<oscPtr *>( _n->m_pluginData )->oscLeft = oscs_l[0];
-		static_cast<oscPtr *>( _n->m_pluginData )->oscRight = oscs_r[0];
+		static_cast<oscPtr *>( _n->m_pluginData )->oscLeft = &oscs_l[0];
+		static_cast<oscPtr *>( _n->m_pluginData )->oscRight = &oscs_r[0];
```
Comment (lukas-w) [inline plugins/organic/organic.cpp]: `&oscs_l[0]` and `&oscs_r[0]` can be written as `oscs_l` and `oscs_r` respectively.

File: plugins/organic/organic.cpp [hunk 6]
```diff
@@ -325,11 +326,7 @@ void organicInstrument::playNote( NotePlayHandle * _n,
 
 void organicInstrument::deleteNotePluginData( NotePlayHandle * _n )
 {
-	delete static_cast<Oscillator *>( static_cast<oscPtr *>(
-						_n->m_pluginData )->oscLeft );
-	delete static_cast<Oscillator *>( static_cast<oscPtr *>(
-						_n->m_pluginData )->oscRight );
-	
+	MM_FREE( static_cast<oscPtr *>(_n->m_pluginData )->oscLeft);
```
Comment (lukas-w) [inline plugins/organic/organic.cpp]: In analogy to the allocation, this should be written as
```cpp
delete[] static_cast<oscPtr *>(_n->m_pluginData )->oscLeft
```

Using `MM_FREE` won't call the object's destructors.

File: plugins/triple_oscillator/TripleOscillator.cpp [hunk 7]
```diff
@@ -300,23 +300,24 @@ void TripleOscillator::playNote( NotePlayHandle * _n,
 {
```
Comment (lukas-w) [inline plugins/triple_oscillator/TripleOscillator.cpp]: Same comments apply to `TripleOscillator.cpp`.

**Comment by lukas-w** [comment]
Closing as there's been no reaction to the review and #3873 already mitigates these performance issues greatly.

**Pull request closed without merging**